File: oop/labaratories/labaratory-work-2/Domain/Dish.cs
namespace labaratory_work_2.Domain;

public abstract class Dish(string name, int weight = 100, double price = 10.0)
{
    public Guid Id { get; } = Guid.NewGuid();
    public string Name { get; set; } = name;
    public int Weight { get; set; } = weight;
    public double Price { get; set; } = price;

    public Dish(string name) : this(name, 200, 20.0)
    {
    }

    public Dish() : this("Dish")
    {
    }

    public abstract string DisplayInfo();
    public abstract double CalcFullPrice();
}

File: oop/lab-3/src/main/java/com/example/demo/domain/Dish.java
package com.example.demo.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "dishes")
@Inheritance(strategy = InheritanceType.JOINED)
@Getter @Setter
public abstract class Dish {
    @Id
    @GeneratedValue
    private UUID id;
    private String name;
    private int weight;
    private double price;

    protected Dish() {}

    public Dish(String name, int weight, double price) {
        this.name = name;
        this.weight = weight;
        this.price = price;
    }

    public Dish(String name) {
        this(name, 200, 20.0);
    }

    public abstract String displayInfo();
    public abstract double calcFullPrice();
}